gridmenuview/src/main/java/com/bcc/gridmenuview/model/MenuItem.java

```diff
@@ -2,20 +2,22 @@
 
 import android.graphics.drawable.Drawable;
 
+import com.bcc.gridmenuview.provider.ImageProvider;
+
 public class MenuItem {
     private String title;
-    private Drawable image;
+    private ImageProvider imageProvider;
 
-    public MenuItem(String title, Drawable image){
+    public MenuItem(String title, ImageProvider imageProvider) {
         this.title = title;
-        this.image = image;
+        this.imageProvider = imageProvider;
```

Comment:
Can we keep the previous constructor? That way, we don't introduce breaking-change to the previous users.

Do something like this:
```java
public MenuItem(String title, Drawable image) {
    this(title, new DrawableImageProvider(image));
}
```

Reply:
added mas @gifff , please check, thanks.